src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -479,14 +522,157 @@ void shouldDisplaySeriesPosition_withSeriesPositionPopulated_whenBookHasSeriesPo
     @Test
     void shouldNotDisplaySeriesPosition_whenBookDoesNotHaveSeriesPosition() {
         // given
-        bookForm = createBookForm(false);
+        bookForm = createBookForm(READ, false);
 
         bookForm.openForm();
 
         // then
         Assertions.assertFalse(bookForm.seriesPositionFormItem.isVisible());
     }
 
+    /**
+     * Tests whether a book is saved to the database when the save-event is called
+     */
+    @Test
+    void shouldSaveBookToDatabase() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+
+        // when
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book bookInDatabase = bookService.findAll().get(0);
+        assertBookInSpecificShelf(TO_READ, bookInDatabase);
+    }
+
+    /**
+     * Tests whether a book is updated in the database when the save-event is called
+     */
+    @Test
+    void shouldUpdateValuesForBook() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        Book savedBook = bookService.findAll().get(0);
+        populateBookFormWithExistingBook(READ, savedBook);
+
+        // when
+        bookForm.bookTitle.setValue("IT");
+        bookForm.authorFirstName.setValue("Stephen");
+        bookForm.authorLastName.setValue("King");
+        bookForm.bookGenre.setValue(Genre.HORROR);
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book updatedBook = bookService.findAll().get(0);
+        Assertions.assertEquals("IT", updatedBook.getTitle());
+        Assertions.assertEquals(READ, updatedBook.getPredefinedShelf().getPredefinedShelfName());
+        Assertions.assertEquals("Stephen", updatedBook.getAuthor().getFirstName());
+        Assertions.assertEquals("King", updatedBook.getAuthor().getLastName());
+        Assertions.assertEquals(Genre.HORROR, updatedBook.getGenre());
+        Assertions.assertEquals(dateStarted, updatedBook.getDateStartedReading());
+        Assertions.assertEquals(dateFinished, updatedBook.getDateFinishedReading());
+    }
+
+
+    private static Stream<Arguments> shelfCombinations() {
+        return Stream.of(
+                Arguments.of(TO_READ, READING),
+                Arguments.of(TO_READ, READ),
+                Arguments.of(TO_READ, DID_NOT_FINISH),
+                Arguments.of(READING, DID_NOT_FINISH),
+                Arguments.of(READING, READ)
+                // TODO: these cases are not passing atm because of issue #271
```

Comment:
Could you expand `atm` in case someone else doesn't know what that means?

Reply:
done